ERCS/erc-7754.md

```diff
@@ -0,0 +1,256 @@
+  "type": "object",
+  "properties": {
+    "publicKeys": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "id": { "type": "string" },
+          "alg": { "type": "string" },
+          "publicKey": { "type": "string" }
+        }
+      }
+    }
+  }
+}
+```
+
+#### RPC method
+
+The parameters of `wallet_signedRequest` are specified by this TypeScript interface:
+
+```typescript
+type RequestPayload<Params> = { method: string; params: Params };
+
+type SignedRequestParameters<Params> = [
+  requestPayload: RequestPayload<Params>,
+  signature: `0x${string}`, // TODO: Is this the format we want?
+  keyId: string,
+];
+```
+
+Here's a non-normative example of calling `wallet_signedRequest` using the [EIP-1193](./eip-1193.md) provider interface:
+
+```typescript
+const keyId = '1';
+const requestPayload: RequestPayload<TransactionParams> = {
+  method: 'eth_sendTransaction',
+  params: [
+    {
+      /* ... */
+    },
+  ],
+};
+const signature: `0x${string}` = await getSignature(requestPayload, keyId);
+
+// Using the EIP-1193 provider interface
+const result = await ethereum.request({
+  method: 'wallet_signedRequest',
+  params: [requestPayload, signature, keyId],
+});
+```
+
+#### Signature verification
+
+1. Upon receiving an [EIP-1193](./eip-1193.md) call, the wallet MUST check of the existence of the TWIT manifest for the `sender.tab.url` domain
+   a. The wallet MUST verify that the manifest is hosted on the `sender.tab.url` domain
+   b. The wallet SHOULD find the DNS TXT record to find the manifest location
+   b. The wallet MAY first try the `/.well-known/twit.json` location
+2. If TWIT is NOT configured for the `sender.tab.url` domain, then proceed as usual
+3. If TWIT is configured and the `request` method is used, then the wallet SHOULD display a visible and actionable warning to the user
+   a. If the user opts to ignore the warning, then proceed as usual
+   b. If the user opts to cancel, then the wallet MUST cancel the call
+4. If TWIT is configured and the `wallet_signedRequest` method is used with the parameters `requestPayload`, `signature` and `keyId` then:
+   a. The wallet MAY display a visible cue indicating that this interaction is signed
+   b. The wallet MUST verify that the keyId exists in the TWIT manifest and find the associated key record
+   c. From the key record, the wallet MUST use the `alg` field and the `publicKey` field to verify `requestPayload` integrity by calling `crypto.verify(alg, key, signature, requestPayload)`
+   d. If the signature is invalid, the wallet MUST display a visible and actionable warning to the user
+   i. If the user opts to ignore the warning, then proceed to call `request` with the argument `requestPayload`
+   ii. If the user opts to cancel, then the wallet MUST cancel the call
+   e. If the signature is valid, the wallet MUST call `request` with the argument `requestPayload`
+
+#### Example method implementation (wallet)
```

Comment:
Should probably move this to the Reference Implementation section and link to it there from here.

Reply:
It's up to you. It's short enough that it's probably fine to stay where it is.